### PR TITLE
HYDRA-178 adding a window plugin

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

File: core-plugins/src/main/java/co/cask/hydrator/plugin/streaming/Window.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.etl.api.streaming.Windower;
+
+/**
+ * Window plugin.
+ */
+@Plugin(type = Windower.PLUGIN_TYPE)
+@Name("Window")
+@Description("Windows a pipeline.")
+public class Window extends Windower {
```
Comment (anwar6953) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/streaming/Window.java:30]: just curious - why is there a Windower api? What are some other types of windows? (perhaps some not based upon time, but is that even supported by the API?)

Comment (albertshau) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/streaming/Window.java:30]: you could think of one that reads the windows from some other api. But the real reason is that it lets us treat windowers just like any other plugin in a pipeline

File: core-plugins/docs/Window-windower.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            "properties":{
+                "width": "5",
+                "slideInterval": "2"
+            }
+        }
+    }
+
+With this setup, the plugin would generate the following windowed batches:
+
+
```
Comment (awholegunch) [inline core-plugins/docs/Window-windower.md:44]: remove line

File: core-plugins/docs/Window-windower.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+--------
+Windows are used when you wish to group smaller batches of events into larger batches of events.
+For example, if your streaming pipeline has a batch interval of one second, you may want to accumulate
+thirty seconds of events before running an aggregation, in order to accumulate enough events to generate
+useful aggregates. You may also want to slide the window at a slower frequency. For example, you may want to
+calculate those aggregates over every five seconds instead of every second.
+
+Properties
+----------
+**width:** the width of the window in seconds. Must be a multiple of the batch interval.
```
Comment (awholegunch) [inline core-plugins/docs/Window-windower.md]: The width of the window...

File: core-plugins/docs/Window-windower.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+Each window generated will contain all the events from this many seconds.
+
+**slideInterval:** the sliding interval of the window in seconds. Must be a multiple of the batch interval.
```
Comment (awholegunch) [inline core-plugins/docs/Window-windower.md]: The sliding interval..

File: core-plugins/docs/Window-windower.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            "name": "Window",
+            "type": "windower",
+            "properties":{
+                "width": "5",
+                "slideInterval": "2"
+            }
+        }
+    }
+
+With this setup, the plugin would generate the following windowed batches:
```
Comment (awholegunch) [inline core-plugins/docs/Window-windower.md]:
```
With this setup, the plugin would [could?] generate these windowed batches:
```

File: core-plugins/docs/Window-windower.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+Description
+-----------
+The Window plugin is used to window a part of a streaming pipeline.
+
+
+Use Case
+--------
+Windows are used when you wish to group smaller batches of events into larger batches of events.
+For example, if your streaming pipeline has a batch interval of one second, you may want to accumulate
```
Comment (awholegunch) [inline core-plugins/docs/Window-windower.md:12]: Where is the batch interval set?

Comment (awholegunch) [inline core-plugins/docs/Window-windower.md:12]: I don't see that in our documentation anywhere.

Comment (albertshau) [inline core-plugins/docs/Window-windower.md:12]: batch interval is set on a pipeline level. It's like 'instances' for cdap-etl-realtime or 'schedule' for cdap-data-pipeline. I still need to add docs for cdap-data-streams.

**Comment by sreevatsanraman** [comment]
LGTM

**Pull request merged**